### PR TITLE
Specify the actual size of the speaker icon images to reduce memory usage

### PR DESCRIPTION
**Pull request by hkurokawa**
## Related Issue
#89 Specify the actual image size to reduce memory usage

## Contents
- [x] Change `DataBindingAttributeUtil#setSpeakerImageUrl` so that it accepts the size of the image on the screen
- [x] Resize the image with the given actual size
- [x] Update all the `ImageView` elements in the layout XML files using `app:speakerImageUrl` attribute so that they specify the actual image size on the screen

#### Before
droidkaigiclient_java_-_droidkaigi2016_-____git_github_com_konifar_droidkaigi2016__-_android_studio_2_0_beta_2

#### After
databindingattributeutil_java_-_droidkaigi2016_-____git_github_com_konifar_droidkaigi2016__-_android_studio_2_0_beta_2_2

[Review on Reviewable](https://reviewable.io/reviews/konifar/droidkaigi2016/94)

### CODE REVIEW

**Comment by konifar** [comment]
:eyes: 
Cool benchmark! I'll check it. But this PR is conflicted, could you merge? Sorry.

**Comment by hkurokawa** [comment]
@konifar Thank you for your pointing it out.  I have just merged the latest head of the master branch.  Could you review that?  Thanks!

File: app/src/main/java/io/github/droidkaigi/confsched/util/DataBindingAttributeUtil.java [hunk 1]
```diff
@@ -22,15 +22,16 @@
 
 public class DataBindingAttributeUtil {
 
-    @BindingAdapter("speakerImageUrl")
-    public static void setSpeakerImageUrl(ImageView imageView, @Nullable String imageUrl) {
+    @BindingAdapter({"speakerImageUrl", "speakerImageSize"})
+    public static void setSpeakerImageUrlWithSize(ImageView imageView, @Nullable String imageUrl, float sizeInDimen) {
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/DataBindingAttributeUtil.java]: Wow, I don't know this way! It looks good :+1:

**Comment by hotchemi** [comment]
:+1:

File: app/src/main/java/io/github/droidkaigi/confsched/util/DataBindingAttributeUtil.java [hunk 2]
```diff
@@ -22,15 +22,16 @@
...
         if (TextUtils.isEmpty(imageUrl)) {
             imageView.setImageDrawable(ContextCompat.getDrawable(imageView.getContext(), R.drawable.ic_speaker_placeholder));
         } else {
+            final int size = (int)(sizeInDimen + 0.5f);
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/DataBindingAttributeUtil.java]: What is `0.5f` ? Is this border line size?

Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched/util/DataBindingAttributeUtil.java]: Maybe `imageView.getContext().getResources().getDimension(R.dimen.line_small)` is better, right?

Comment (hkurokawa) [inline app/src/main/java/io/github/droidkaigi/confsched/util/DataBindingAttributeUtil.java]: No, it is just for rounding up the float value to int.  I read the similar code in `TypedValue#complextToDimensionPixelSize()` as below.

```
    {
        final float value = complexToFloat(data);
        final float f = applyDimension(
                (data>>COMPLEX_UNIT_SHIFT)&COMPLEX_UNIT_MASK,
                value,
                metrics);
        final int res = (int)(f+0.5f);
        if (res != 0) return res;
        if (value == 0) return 0;
        if (value > 0) return 1;
        return -1;
    }
```

But maybe it is confusing and we should use resource ID, instead.

**Comment by konifar** [comment]
LGTM!

**Comment by hkurokawa** [comment]
I am merging this!

**Comment by konifar** [comment]
@hkurokawa Could you delete this branch?

**Comment by hkurokawa** [comment]
@konifar Sorry. Just deleted.

**Pull request merged**